src/main/java/stelnet/commodity/market/MarketRepository.java

```diff
@@ -0,0 +1,46 @@
+package stelnet.commodity.market;
+
+import java.util.HashMap;
+import java.util.List;
+
+import stelnet.commodity.CommodityTab;
+
+public class MarketRepository {
+
+    HashMap<String, List<MarketApiWrapper>> buyMarkets = new HashMap<>();
+    HashMap<String, List<MarketApiWrapper>> sellMarkets = new HashMap<>();
+    private static MarketRepository instance;
```

Comment:
Problem with statics is they will stay between loads / new game. Given long enough playtime with many resets and reloads this will pollute memory and possibly cause massive slowdown through constant garbage collection attempts.

Read below comment as well.

Reply:
One way would be to create the instance of MarketRepository in CommodityBoard constructor and dependency inject it to factory classes. However, I am unsure how this `IntelHelper.getFirstIntel(CommodityBoard.class)` really behaves,

As the MarketRepository instance should be re-created every time we access commodities.

Reply:
Added TODO to investigate.